linux/hid.c

```diff
@@ -493,127 +495,229 @@ parse_uevent_info(const char *uevent, unsigned *bus_type,
 }
 
 
-static int get_device_string(hid_device *dev, enum device_string_id key, wchar_t *string, size_t maxlen)
+static void get_device_info_for_device(struct udev_device *raw_dev, unsigned short vendor_id, unsigned short product_id, struct hid_device_info ** first_dev, struct hid_device_info ** last_dev)
```

Comment:
this looks unnecessarily complicated, I expect to have something like:
`static struct hid_device_info * get_device_info_for_device(struct udev_device *raw_dev)`

Reply:
maybe even: `create_device_info_for_device` - note: *create_*, not *get_*

Reply:
I ended up with this because it can return any number of devices. So it was either this, or immediately afterwards it would have to loop through the returned device to find `last_dev`. It didnt feel good to be iterating through a return value like that.
```
struct hid_device_info * last_dev = first_dev->next;
while (last_dev->next) {
  last_dev = last_dev->next;
}
```
Something like that. I did do this initially, but after it got into an infinite loop a couple of times I decided to change it.

I suppose `first_dev` could be returned instead of being passed out like this, but this felt a bit better for consistency.